csharp/src/Google.Protobuf/Reflection/EnumDescriptor.cs

```diff
@@ -129,7 +129,7 @@ public EnumValueDescriptor FindValueByName(string name)
         /// The (possibly empty) set of custom options for this enum.
         /// </summary>
         [Obsolete("CustomOptions are obsolete. Use GetOption")]
-        public CustomOptions CustomOptions => new CustomOptions(Proto.Options?._extensions?.ValuesByNumber);
+        public CustomOptions CustomOptions => Proto?.Options?.CreateCustomOptions() ?? CustomOptions.Empty;
```

Comment:
AFAICT If `Proto` is null, it would mean that something is badly broken, so I don't quite understand the `Proto?.` operator (it's also something you didn't have in the previous version of the code).
Same for all `*Descriptor` files.